src/Catty/ViewController/Download/ProjectDetailStoreViewController.h

```diff
@@ -27,12 +27,15 @@
 #import "BaseTableViewController.h"
 
 @class CatrobatProject;
+@class StoreProjectDownloader;
+@class Data;
 
 @interface ProjectDetailStoreViewController : UIViewController<ProjectStoreDelegate, UIScrollViewDelegate, CBFileManagerDelegate, TTTAttributedLabelDelegate, NSURLConnectionDataDelegate, UIGestureRecognizerDelegate>
 
 @property (nonatomic, strong) CatrobatProject *project;
 @property (nonatomic, strong) NSMutableDictionary *projects;
 @property (nonatomic, weak) IBOutlet UIScrollView *scrollViewOutlet;
+@property (nonatomic, strong) StoreProjectDownloader *storeProjectDownloader;
```

Comment:
It is a very good idea to make this dependency substitutable for testing purposes (e.g., to be replaced with a test double). However, I would recommend to do that directly in the Swift extension, as a computed property. By doing so we do not have to make this Swift class available for ObjectiveC.

Reply:
Thank you, I've changed that!